src/PeriodEnd/TransferPayments/SFA.DAS.ProviderPayments.Calc.TransferPayments.UnitTests/ServiceTests/WhenProcessingTransfers.cs

```diff
@@ -0,0 +1,281 @@
+                    var actual = result.TransferPayments.Sum(x => x.Amount);
+
+                    actual.Should().Be(expected);
+                }
+
+                [Test, TransfersAutoData]
+                public void ThenTheTransferAmountsAreCorrect(
+                    LevyTransferService sut,
+                    List<RequiredTransferPayment> requiredPayments,
+                    Account receiver
+                )
+                {
+                    var entity = new DasAccount { Balance = 1000000, TransferBalance = 1000000, IsLevyPayer = true };
+                    var sender = new Account(entity);
+                    var expected = requiredPayments.Sum(x => x.AmountDue);
+
+                    var result = sut.ProcessTransfers(sender, receiver, requiredPayments);
+                    var actual = result.AccountTransfers.Sum(x => x.Amount);
+
+                    actual.Should().Be(expected);
+                }
+            }
+
+            public class AndThereIsNotSufficientBalanceForAllPayments
+            {
+                [Test, TransfersAutoData]
+                public void ThenThePaymentAmountsAreCorrect(
+                    LevyTransferService sut,
+                    List<RequiredTransferPayment> requiredPayments,
+                    Account receiver
+                )
+                {
+                    var expected = requiredPayments.Sum(x => x.AmountDue) / 2;
+
+                    var entity = new DasAccount { Balance = expected, TransferBalance = 1000000, IsLevyPayer = true };
+                    var sender = new Account(entity);
+                    
+                    var result = sut.ProcessTransfers(sender, receiver, requiredPayments);
+                    var actual = result.TransferPayments.Sum(x => x.Amount);
+
+                    actual.Should().Be(expected);
+                }
+
+                [Test, TransfersAutoData]
+                public void ThenTheTransferAmountsAreCorrect(
+                    LevyTransferService sut,
+                    List<RequiredTransferPayment> requiredPayments,
+                    Account receiver
+                )
+                {
+                    var expected = requiredPayments.Sum(x => x.AmountDue) / 2;
+
+                    var entity = new DasAccount { Balance = expected, TransferBalance = 1000000, IsLevyPayer = true };
+                    var sender = new Account(entity);
+                    
+                    var result = sut.ProcessTransfers(sender, receiver, requiredPayments);
+                    var actual = result.AccountTransfers.Sum(x => x.Amount);
+
+                    actual.Should().Be(expected);
+                }
+            }
+        }
+
+        public class AndTheTransferBalanceDoesNotCoverAllPayments
+        {
+            [Test, Ignore("To keep structure in the results")]
+            public void Null() { }
+
+            public class AndThereIsSufficientBalanceForAllPayments
```

Comment:
AndTheTransferBalanceDoesNotCoverAllPayments.AndThereIsSufficientBalanceForAllPayments

doesn't make sense to me but could just be my misunderstanding